### PR TITLE
Check that network exists when adding a custom network

### LINKED ISSUE #3800: Check that network exist when adding a custom network
### **Description**
Before adding a custom network, check that this network isn't already added

### **Acceptance Criteria**
-   (Stretch UX Improvement-Create ticket if not completed in this ticket) If network already added and the user attempts to add custom network again check and prompt the user that it is already added (check against chain ID)

#3545

### PR DESCRIPTION
**Pull request by blackdevelopa**
**Description**
Before adding a custom network, check that this network isn't already added (check against chain ID)

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #3800

### CODE REVIEW

File: app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js [hunk 1]
```diff
@@ -243,6 +243,20 @@ class NetworkSettings extends PureComponent {
 		return true;
 	};
 
+	checkIfChainIdExists = async (chainId) => {
+		const checkCustomNetworks = this.props.frequentRpcList.filter((item) => item.chainId === chainId);
```
Comment (sethkfman) [inline app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js:247]: Per the demo today. Confirm with @rekmarks that we should use RpcUrl as the uniqueness for a network instead of chainId

**Review by sethkfman — COMMENTED** [review]
Let me know when you update the uniqueness check.

File: app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js (hunk 1 above)
Comment (rekmarks) [reply app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js:247]: For the form in network settings, we should use the RPC URL for uniqueness, **not** the chain ID.

Historically, the extension has used the RPC URL as the unique identifier in the form, and the chain ID for `wallet_addEthereumChain`. This distinction was made under the assumption that dapps should not care what the RPC URL is, and letting dapps suggest that the user adds a chain that already exists in their network list could create a confusing and potentially dangerous user experience. We retained the ability to add multiple RPC URLs for the same endpoint via the form for the benefit of advanced users and developers.

I believe that the extension team has re-evaluated some of these choices recently, but I'm not the right person to speak to that. I'd check in with @danjm for more details on what the extension is planning.

To recap:
- Use the RPC URL as the unique identifier for chains. Ensure that all RPC URLs are normalized (e.g., no trailing slashes) so that they can be reliably used for this purpose.
    - I'm not sure if you should treat the same URL with different protocols (i.e. `http` and `https`) as the same or distinct. I'd check what the extension does.
    - Keep in mind that a user may need to be able to use all components of a URL, including the port, query parameters, etc.

You can find the extension's validation logic for the RPC URL here: https://github.com/MetaMask/metamask-extension/blob/develop/ui/pages/settings/networks-tab/networks-form/networks-form.js/#L374-L412

File: app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js [hunk 2]
```diff
@@ -243,6 +243,20 @@ class NetworkSettings extends PureComponent {
 		return true;
 	};
 
+	checkIfNetworkExists = async (rpcUrl) => {
+		const checkCustomNetworks = this.props.frequentRpcList.filter((item) => item.rpcUrl === rpcUrl);
+		if (checkCustomNetworks.length > 0) {
+			this.setState({ warningRpcUrl: strings('app_settings.chain_id_exists') });
```
Comment (sethkfman) [inline app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js:249]: I would update the warning message as well so it is not chain ID

**Review by sethkfman — COMMENTED** [review]
Copy change required

File: app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js (hunk 2 above)
Comment (blackdevelopa) [reply app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js]: Updated

**Review by sethkfman — APPROVED** [review]
LGTM

**Review by cortisiko — APPROVED** [review]
🌮 🌮 🌮
